# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/base/nsStandardURL.h
# Commit: ef6d81f0b298
# Full Hash: ef6d81f0b2989624a9918ec5d790dc2745db20ef
# Author: Valentin Gosu <valentin.gosu@gmail.com>
# Date: 2018-11-15 09:52:43
# Regressor Bug: 1495313
# File Overlap Count: 1
# Description:
#   Bug 1495313 - Allow password with empty username in URLs r=dragana
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D11254
# ==============================================================================

diff -r 06b12fd41ff1 -r ef6d81f0b298 netwerk/base/nsStandardURL.h
--- a/netwerk/base/nsStandardURL.h	Wed Nov 14 19:37:42 2018 +0000
+++ b/netwerk/base/nsStandardURL.h	Wed Nov 14 19:10:42 2018 +0000
@@ -534,12 +534,18 @@
 nsStandardURL::Userpass(bool includeDelim)
 {
     uint32_t pos=0, len=0;
-    // if there is no username, then there can be no password
-    if (mUsername.mLen > 0) {
-        pos = mUsername.mPos;
-        len = mUsername.mLen;
-        if (mPassword.mLen >= 0)
-            len += (mPassword.mLen + 1);
+    if (mUsername.mLen > 0 || mPassword.mLen > 0) {
+        if (mUsername.mLen > 0) {
+            pos = mUsername.mPos;
+            len = mUsername.mLen;
+            if (mPassword.mLen >= 0) {
+                len += (mPassword.mLen + 1);
+            }
+        } else {
+            pos = mPassword.mPos - 1;
+            len = mPassword.mLen + 1;
+        }
+
         if (includeDelim)
             len++;
     }